### PR TITLE
add missing collection.reload documentation [ci skip]

### PR DESCRIPTION
**Pull request by yukideluxe**
Hello!

I missed some documentation that I think is nice to have when I did https://github.com/rails/rails/pull/30125 and I noticed about this when I did the backport to 5-0-stable https://github.com/rails/rails/pull/30141

As promised, here is the extra bit 😄 

Sorry for the noise!

### CODE REVIEW

File: activerecord/lib/active_record/associations.rb [hunk 1]
```diff
@@ -1250,6 +1250,9 @@ module ClassMethods
         # [collection.create!(attributes = {})]
         #   Does the same as <tt>collection.create</tt>, but raises ActiveRecord::RecordInvalid
         #   if the record is invalid.
+        # [collection.reload]
+        #   Returns an array of all of the associated objects, forcing a database read.
```
Comment (sgrif) [inline activerecord/lib/active_record/associations.rb]: Does this return an array? I'm pretty sure it just returns `self`

File: activerecord/lib/active_record/associations.rb [hunk 2]
```diff
@@ -1250,6 +1250,9 @@ module ClassMethods
...
+        #   An empty array is returned if none are found.
```
Comment (sgrif) [inline activerecord/lib/active_record/associations.rb]: Same here -- I don't think this returns an array.

**Review by sgrif — CHANGES_REQUESTED** [review]

File: activerecord/lib/active_record/associations.rb (hunk 1 above)
Comment (yukideluxe) [inline activerecord/lib/active_record/associations.rb]: Hi @sgrif, thanks for the feedback!

I based this on this documentation above https://github.com/yukideluxe/rails/blob/master/activerecord/lib/active_record/associations.rb#L1194-L1196

Both `collection` and `collection.reload` return an `ActiveRecord::Associations::CollectionProxy` so they behave the same. Does it mean that the documentation for `#collection` is also wrong? I'm not sure how to rephrase it, though 🙈

Comment (yukideluxe) [inline activerecord/lib/active_record/associations.rb]: updated the link, sorry :)

Comment (sgrif) [inline activerecord/lib/active_record/associations.rb]: Yes, that documentation is also wrong. It was last touched by https://github.com/rails/rails/commit/6e754551 9 years ago when the behavior was different. I would rephrase it by replacing `array` with `Relation`

Comment (sgrif) [inline activerecord/lib/active_record/associations.rb]: You could also just link to `Relation#reload`

Comment (yukideluxe) [inline activerecord/lib/active_record/associations.rb]: sure! Let me do the changes and I'll amend this commit! Thanks for the tip! :)

Comment (yukideluxe) [inline activerecord/lib/active_record/associations.rb]: @sgrif done here and in the backport https://github.com/rails/rails/pull/30141! ✌️

**Comment by tjschuck** [comment]
@sgrif The (now merged) backport @yukideluxe mentioned (#30141) was to `5-0-stable` — this should also probably be backported to `5-1-stable`.
